ow_faults/src/FaultInjector.cpp

```diff
@@ -75,6 +81,35 @@ void FaultInjector::setFaultsMessage(ow_faults::PTFaults& msg, ComponentFaults v
   msg.value = static_cast<uint>(value); //should be HARDWARE for now
 }
 
+void FaultInjector::antennaePanFaultCb(const std_msgs::Float64& msg){
+  std_msgs::Float64 pan_msg;
+  if (m_faults.ant_pan_encoder_failure || m_faults.ant_pan_torque_sensor_failure) {
+    if (isnan(faultPanValue)){
+      faultPanValue = msg.data;
+    }
+    pan_msg.data = faultPanValue;
+    m_fault_ant_pan_pub.publish(pan_msg);
+  } else {
+    //turning off fault, should revert to most recent msg val
+    m_fault_ant_pan_pub.publish(msg);
+    faultPanValue = msg.data;
+  }
+}
+
+void FaultInjector::antennaeTiltFaultCb(const std_msgs::Float64& msg){
```

Comment:
I realize that this is functionally the same as the antennaePanFaultCb(). The difference is the we are checking either the pan or tilt fault selection/variable. Is there a way in a callback to find out which topic it was subscribed to? If so I can condense these functions.

Reply:
I tried to look into boost::bind for the subscriber, but running into trouble. I'm seeing a lot of references initialize subscribers outside of the class in a main, but for our purposes, the subscriber is initialized in the class initialization. Would love any pointers.

Reply:
The only modern technique I know to refactor these functions into one is a template function parameterized over the classes involved, and reference arguments (&) for the float-valued member variables that are being set.   The resulting signature and calls are a bit longer, but I think this should be done.  (The old-school C technique is a macro, which I don't recommend).

Reply:
You can factor out the shared code into a private method and then invoke the shared private method from each call back. for example:
```suggestion
void FaultInjector::private_shared_method(const std_msgs::Float64& msg, bool flag1, bool flag2, float& m_value, Publisher m_publisher){
std_msgs::Float64 out_msg;
  if (flag1 || flag2) {
    if (isnan(m_value)){
      m_value = msg.data;
    }
    out_msg.data = m_value;
    m_publisher.publish(out_msg);
  } else {
    m_publisher.publish(msg);
    m_value = msg.data;
  }
}
```
Then each callback would invoke the `private_shared_method`

Reply:
I like this, and using the base class Publisher as parameter is much better than template.

Reply:
updated! TY for the suggestions!